cmake/configure/configure_1_threads.cmake

```diff
@@ -38,6 +38,16 @@ MACRO(SETUP_THREADING)
 
     RESET_CMAKE_REQUIRED()
 
+    #
+    # The FindThreads macro returned a linker option instead of the actual
+    # library name in earlier versions. We still require the linker option,
+    # so we fix the corresponding variable.
+    #  - See: https://gitlab.kitware.com/cmake/cmake/issues/19747
+    #
+    IF(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.16.0")
+      LIST(TRANSFORM CMAKE_THREAD_LIBS_INIT PREPEND "-l")
+    ENDIF()
```

Comment:
Would you mind changing this to `REGEX_REPLACE` a match `^pthreads$` with `-lpthreads`? That way you don't need the version check and we avoid accidentally producing `-l-lpthreads` the moment something changes in a future CMake version.

Reply:
*mhm* What about you just check for your expectation then?
```
IF("${CMAKE_THREAD_LIBS_INIT}" MATCHES "^pthread$")
  STRING(PREPEND CMAKE_THREAD_LIBS_INIT "-l")
ENDIF()
```